### DISCUSSION TITLE
feature: fallback value for the env shortcode

### DISCUSSION THREAD

**Discussion by aronatkins** [quarto-dev/quarto-cli#8315] [Feature Requests]
### Description

The `env` shortcode does not allow a fallback value.

In R, we might:

```r
Sys.getenv("FRUIT", "apples")
```

In Python, we might:

```python
os.environ.get("FRUIT", "apples")
```

In simple Quarto documents (using the Markdown engine), we do not have either of these options. If the document were accompanied by a `_quarto.yml` file (and seen as a "project"), that document could use an `_environment` file to provide default environment values. The `_environment` file is only processed for projects, not standalone documents.

Proposal: Extend the `env` shortcode to allow a fallback value.

```markdown
(awesome website)[{{< env TARGET_URL "https://quarto.org/" >}}]
```

**Comment by mcanouil** [comment]
Thanks.
Closing in favour of:
- https://github.com/quarto-dev/quarto-cli/issues/8316
